nixos/modules/services/networking/webhook.nix

```diff
@@ -0,0 +1,189 @@
+        example = literalExample "pkgs.webhook";
+        description = ''
+          Webhook package to use.
+        '';
+      };
+      user = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = ''
+          Webhook will be run under this user (user will be created if it doesn't exist. This can be your user name).
+        '';
+      };
+      group = mkOption {
+        type = types.str;
+        default = defaultUser;
+        description = ''
+          Webhook will be run under this group (group will be created if it doesn't exist. This can be your group name).
+        '';
+      };
+      hotreload = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Watch hooks file for changes and reload them automatically";
+      };
+      ip = mkOption {
+        type = types.str;
+        default = "0.0.0.0";
+        description = "ip the webhook should serve hooks on";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 9000;
+        description = "port the webhook should serve hooks on";
+      };
+      template = mkOption {
+        type = types.bool;
+        default = false;
+        description = "parse hooks file as a Go template";
+      };
+      urlprefix = mkOption {
+        type = types.str;
+        default = "hooks";
+        description = "url prefix to use for served hooks (protocol://yourserver:port/PREFIX/:hook-id)";
+      };
+      hooks = mkOption {
+        default = {};
+        description = "Hooks the webhook should serve";
+        example = {
+          redeploy-webhook = {
+            id = "redeploy-webhook";
+            execute-command = "/var/scripts/redeploy.sh";
+            command-working-directory = "/var/webhook";
+          };
+        };
+        type = types.attrsOf (types.submodule ({ config, ... }: {
+          options = {
+            id = mkOption {
+              type = types.str;
+              default = config._module.args.name;
+              description = "specifies the ID of your hook. This value is used to create the HTTP endpoint (http://yourserver:port/hooks/your-hook-id)";
+            };
+            execute-command = mkOption {
+              type = types.str;
+              description = "specifies the command that should be executed when the hook is triggered";
+            };
+            command-working-directory = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = "specifies the working directory that will be used for the script when it's executed";
+            };
+            response-message = mkOption {
+              type = types.nullOr types.str;
+              default = null;
+              description = "specifies the string that will be returned to the hook initiator";
+            };
+            response-headers = mkOption {
+              type = types.nullOr (types.listOf
```

Comment:
`attrsOf str` would be nicer. It's more compact and allows overrides.
If multiple headers of the same name are required, this can be added in a backwards compatible way.